File: programmers/js/level00/120882.js
// 등수 매기기
function solution(score) {
  const scoreAvg = score.map(([eng, math]) => (eng + math) / 2);
  const avgRank = [...scoreAvg].sort((a, b) => b - a);
  
  return scoreAvg.map(score => avgRank.indexOf(score) + 1);
}

// 다른 풀이
// function solution(score) {
//   return score.map((el) => score.filter((v) => (v[0] + v[1]) / 2 > (el[0] + el[1]) / 2).length + 1);
// }

File: programmers/java/level00/120882.java
// 등수 매기기

import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;
import java.util.Comparator;
import java.util.stream.Collectors;

class Solution {
    public int[] solution(int[][] score) {
        List<Double> avgScore = Arrays.stream(score)
            .map(arr -> (arr[0] + arr[1]) / 2.0)
            .collect(Collectors.toList());
        List<Double> sortedScore = new ArrayList<>(avgScore);
        sortedScore.sort(Comparator.reverseOrder());
        
        return avgScore.stream()
            .mapToInt(avg -> sortedScore.indexOf(avg) + 1)
            .toArray();
    }
}